src/main/java/de/hysky/skyblocker/skyblock/item/ItemCooldowns.java

```diff
@@ -17,40 +24,109 @@
+                        .getAsJsonObject("members").entrySet().stream()
+                        .filter(entry -> entry.getKey().equals(playeruuid))
+                        .map(Map.Entry::getValue)
+                        .map(JsonElement::getAsJsonObject)
+                        .findFirst()
+                        .orElseThrow(() -> new IllegalStateException("Player somehow not found inside their own profile!"));
+                for (JsonElement element : players.getAsJsonObject("pets_data").getAsJsonArray("pets")) {
+                    if (!element.getAsJsonObject().get("type").getAsString().equals("MONKEY")) continue;
+                    if (!element.getAsJsonObject().get("active").getAsString().equals("true")) continue;
+                    if (element.getAsJsonObject().get("tier").getAsString().equals("LEGENDARY")) {
+                        CalcMonkeyExp = Double.parseDouble(element.getAsJsonObject().get("exp").getAsString());
+                        MonkeyLevel = 0;
+                        for (int xpLevel : EXPERIENCE_LEVELS) {
+                            if (CalcMonkeyExp < xpLevel) {
+                                break;
+                            } else {
+                                CalcMonkeyExp -= xpLevel;
+                                MonkeyLevel++;
+                            }
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                System.out.println("[Skyblocker] Failed to get Player Pet Data, is the API Down/Limited?");
+            }
+            double BaseCooldown = 2000;
+            double EvolvedAxesCooldownReductionPercentage = MonkeyLevel * 0.5;
+            double MonkeyPetCDRReduction = (BaseCooldown * EvolvedAxesCooldownReductionPercentage) / 100;
+            currentCooldown = BaseCooldown - MonkeyPetCDRReduction;
+        });
     }
 
     public static void afterBlockBreak(World world, PlayerEntity player, BlockPos pos, BlockState state) {
         if (!SkyblockerConfigManager.get().general.itemCooldown.enableItemCooldowns) return;
-
         String usedItemId = ItemUtils.getItemId(player.getMainHandStack());
         if (usedItemId.isEmpty()) return;
-
         if (state.isIn(BlockTags.LOGS)) {
-            if (usedItemId.equals(JUNGLE_AXE_ID)) {
-                if (!isOnCooldown(JUNGLE_AXE_ID)) {
-                    ITEM_COOLDOWNS.put(JUNGLE_AXE_ID, new CooldownEntry(2000));
-                }
-            } else if (usedItemId.equals(TREECAPITATOR_ID)) {
-                if (!isOnCooldown(TREECAPITATOR_ID)) {
-                    ITEM_COOLDOWNS.put(TREECAPITATOR_ID, new CooldownEntry(2000));
+            currentCooldown();
+            if (usedItemId.equals(JUNGLE_AXE_ID) || usedItemId.equals(TREECAPITATOR_ID)) {
+                if (!isOnCooldown(JUNGLE_AXE_ID) || !isOnCooldown(TREECAPITATOR_ID)) {
+                    ITEM_COOLDOWNS.put(usedItemId, new CooldownEntry((int) currentCooldown));
```

Comment:
I think `currentCooldown()` should be moved in the ifs below. It's useless doing an API call if they aren't using a treecapitator